tests/testthat/test-check_crpr.R

```diff
@@ -0,0 +1,58 @@
+library(tibble)
+library(dplyr)
+library(lubridate)
+
+adrs <- tribble(
+  ~USUBJID, ~ADTC,        ~AVALC,
+  "1",      "2020-01-01", "PR",
+  "1",      "2020-02-01", "CR",
+  "1",      "2020-02-16", "NE",
+  "1",      "2020-03-01", "CR",
+  "1",      "2020-04-01", "SD",
+  "2",      "2020-02-06", "PR",
+  "2",      "2020-02-16", "CR",
+  "2",      "2020-03-30", "PR",
+) %>%
+  mutate(
+    ADT = ymd(ADTC),
+    STUDYID = "XX1234"
+  )
+
+# signal_crpr ----
+## Test 1: warning is issued ----
+test_that("signal_crpr Test 1: warning is issued", {
+  expect_warning(signal_crpr(
+    adrs,
+    order = vars(ADT)
+  ),
+  paste("Dataset contains CR records followed by PR.",
+    "Run `get_crpr_dataset()` to access the CR records records followed by PR",
+    sep = "\n"
+  ),
+  fixed = TRUE
+  )
+})
+
+# get_crpr_dataset ----
+## Test 2: dataset if returned ----
+test_that("get_crpr_dataset() Test 2: dataset if returned", {
+  suppress_warning(
+    signal_crpr(adrs,
+      order = vars(ADT)
+    ),
+    "Dataset contains CR records followed by PR"
+  )
+
+  expect_dfs_equal(
+    base = tribble(
+      ~USUBJID, ~ADTC,        ~AVALC,
+      "2",      "2020-02-16", "CR"
```

Comment:
wouldn't it be useful to also keep the subsequent PR record here in the function code so that users easily see the full issue?

Reply:
I have added the PR record.